packages/smooth_app/lib/cache/files/files_cache_impl.dart

```diff
@@ -0,0 +1,123 @@
+// ignore_for_file: avoid_slow_async_io
+part of 'files_cache.dart';
+
+/// Implementation based on [getTemporaryDirectory]
+class _ShortLivingFileCacheManagerImpl extends _FileCacheManagerImpl {
+  _ShortLivingFileCacheManagerImpl._() : super._();
+
+  @override
+  Future<void> _init(String name) async {
+    return _initCache(name, await getTemporaryDirectory());
+  }
+}
+
+/// Implementation based on [getApplicationDocumentsDirectory]
+class _LongLivingFilesCacheManagerImpl extends _FileCacheManagerImpl {
+  _LongLivingFilesCacheManagerImpl._() : super._();
+
+  @override
+  Future<void> _init(String name) async {
+    return _initCache(name, await getApplicationDocumentsDirectory());
+  }
+}
+
+abstract class _FileCacheManagerImpl extends FileCache {
+  factory _FileCacheManagerImpl(FileCacheType type) {
+    switch (type) {
+      case FileCacheType.shortLiving:
+        return _ShortLivingFileCacheManagerImpl._();
+      case FileCacheType.longLiving:
+        return _LongLivingFilesCacheManagerImpl._();
+    }
+  }
```

Comment:
Should not be there.
Here we're supposed to know nothing about the outer world, we're just a file cache repository, that is initialized with a root folder and a subfolder name.
Whoever calls us should set those values.

Reply:
I was thinking of something like that in the "container" `FileCacheManager`:
```dart
late FileCache persistent;
late FileCache temporary;

Future<void> init() { // to be called in main.dart
  persistent = FileCache('files_cache', await getApplicationDocumentsDirectory());
  temporary = FileCache('files_cache', await getTemporaryDirectory());
}
```

Reply:
I've done it a little bit differently. 
Please tell me if it's ok, or if you prefer your solution